packages/react-client/src/ReactFlightClient.js

```diff
@@ -569,6 +569,20 @@ export function parseModelString(
         return BigInt(value.substring(2));
       }
       default: {
+        switch (value) {
```

Comment:
Let's switch on just the first character, like we do for undefined. We can use anything that's not lower-case hexadecimal. I guess negative 0 and negative infinite needs a second if in the `'-'` case.

Reply:
Like shorten to $I and $N? Or do `case 'I': if (value === '$Infinity')`?

Reply:
Or just `case 'I': return Infinity;`

Reply:
```
case 'I': return Infinity
case '-': return value === '-0' ? -0 : -Infinity;
```

This is smaller/faster code on the client (if the switch is optimized at least). It's kind of minimal.

The encoding in the payload/server can be larger to make it easier to debug looking at the payload. That's the thinking behind `$undefined` not just being `$u`. The server code size is less sensitive and the payload size matters less for rare cases.

Reply:
done